### PR TITLE
Add Optional Panic Handler to task group

### PR DESCRIPTION
**Pull request by CPWeaver**
* Add ability to define a panic handler on a task group

### CODE REVIEW

File: taskgroup.go [hunk 1]
```diff
@@ -77,6 +84,21 @@ func (t *TaskGroup) Exec() error {
 	return errorOut
```
Comment (schigh) [inline taskgroup.go:87]: This needs updated so panic handlers can be chained:

Change:
```go
if errorOut == nil && t.Next != nil {
	if t.Next.ErrorHandler == nil {
		t.Next.ErrorHandler = t.ErrorHandler
	}
	t.ensureBag()
	errorOut = t.Next.execWithBag(t.bag)
}
```

to:

```go
if errorOut == nil && t.Next != nil {
	if t.Next.ErrorHandler == nil {
		t.Next.ErrorHandler = t.ErrorHandler
	}
	if t.Next.PanicHandler == nil {
		t.Next.PanicHandler = t.PanicHandler
	}
	t.ensureBag()
	errorOut = t.Next.execWithBag(t.bag)
}
```

Comment (CPWeaver) [reply taskgroup.go:87]: Added, with a test

File: taskgroup_test.go [hunk 2]
```diff
@@ -26,6 +26,11 @@ func failTask(t *TaskGroup) error {
 	return errors.New("fail")
 }
 
+func panicTask(t *TaskGroup) error {
+	fmt.Println("this task will panic")
+	panic("no soup for you")
```
Comment (schigh) [inline taskgroup_test.go:31]: 🥇

**Review by schigh — APPROVED** [review]
💥 awesome 💥

**Pull request merged**